### ISSUE TITLE
Abstract Constructor Assignability not restricted

### ISSUE THREAD

**Issue by aozgaa** [Microsoft/TypeScript#3702]
The following error isn't triggered:

```ts

class A {}

abstract class B extends A {}

var AA : A = B; // assignment should fail.

new AA;
```

In order to fix this, we'll mark the construct signature of the class as abstract and then examine this flag during assignability comparison. Note that when we do assignability comparison in the declaration, we do not examine the construct signature (as we don't require that the extended class' constructors be assignable to the base class).
